### ISSUE TITLE
Provide image URL, pre-process images, correct registration

### ISSUE THREAD

**Issue by drphilmarshall** [linan7788626/LensWranglerPlusPlus#1]
I think we want to be able to pass in an image URL as well as upload a file - many of our images will be on the web.

**Comment by linan7788626** [comment]
Yes. Actually , the default image is loaded in via its URL, so I think this issue will be fixed in next week.

**Comment by drphilmarshall** [comment]
Cool! :-) This will enable Space Warps users to investigate their finds.

**Comment by linan7788626** [comment]
This issue has been fixed. I just added a textbox and a button to upload images via URL (please see the image below).

![screen shot 2016-09-12 at 1 51 10 pm](https://cloud.githubusercontent.com/assets/1016652/18448306/358cae70-78f0-11e6-8855-0720c7ae7032.png)

But it leads to another problem: now I am using 400*400 images in LW++, and angular pixel size is set to be 0.0225 arcsecs. If we want to upload images via URL and model it correctly, we need to preprocess the image first. 
- [ ] preprocessing of the input images (rescaling, and reading in the pixelization of the images).

**Comment by drphilmarshall** [comment]
I see something odd when loading in this image: the "Source Plane" and "Image Plane" panel images are registered differently to the "Mass Model" one, so that the caustics appear in the wrong place.

![image](https://cloud.githubusercontent.com/assets/710903/18481200/373310f2-7990-11e6-921e-4c4e8a5bfa24.png)

(BTW this image illustrates my concern about the "Source Model" panel - the only thing that we should be doing there is controlling the size, ellipticity, and orientation of each source component, and their positions _relative to each other_. The moving cursor in the "Source Plane" panel and the click placed in the "Image Plane" panel should determine the position of the whole source model relative to the lens. I think this means we want the "Source Model" panel to have no image background (which makes sense anyway, because the JPG image belongs in the image plane!), only source components and a central cyan cross. This cross would correspond to the click placed on the "Image Plane" panel (to mark the center of one multiple image) and to the cursor position in the "Source Plane" panel.)

**Comment by linan7788626** [comment]
The odd thing is because of the issue I mentioned in my reply, the picture dimension of the uploaded image is 467_467. However, the dimension of the blocks are 400_400, so we need to preprocess an input image with arbitrary size (for example, rescale it). You can see that the pictures in the top panels are rescaled, but the pictures in the bottom panels are not rescaled, so the center of the lens is shifted. I am working on it now.

In term of the source model, we can only sample one source now, so even though you drew two ellipses, the APP will present the first source only. I am thinking about controlling the sources in a similar way as controlling the lenses.  Controlling the sources separately might be a more sensible way than controlling them together, because of its flexibility.  Please let me know your thoughts about the new strategy. Thanks.

Removing the background image of the source model block makes sense to me, I will try to make it, but if we would like to control sources separately, a cyan cross and caustics might be both necessary.

**Comment by drphilmarshall** [comment]
Agreed - more thoughts in #4. Good idea to stick wth one source for now,
but I think with the image plane positioning we will need the concept of a
"primary source component" to go with that of an image plane "anchor
position". For sanity you may want to disable the drawing of new ellipses
in the source model, and force the user to work with the one they are
given. I can't remember if such an action is easy though :-)

**Comment by linan7788626** [comment]
Disabling the drawing of new ellipses is a sensible option. I will try to make it. 
BTW, I expect that you will recall if it is easy, and instruct me how to do it. :-) Thanks.
